test/IterableWeakMap.spec.ts

```diff
@@ -0,0 +1,282 @@
+                const iwm = new IterableWeakMap()
+                const ref = {}
+                const value = {}
+                const ref2 = {}
+                const value2 = {}
+
+                iwm.set(ref, value)
+                iwm.set(ref2, value2)
+
+                const res = Array.from(iwm.values())
+
+                expect(res).toEqual([value, value2])
+            })
+        })
+    }
+
+    describe('environment supports WeakRef and FinalizationRegistry', () => {
+        runCommonTests()
+
+        describe('memory', () => {
+            let originalFinalizationRegistry: FinalizationRegistry
+            let cleanupMemory = (ref: any) => {}
+
+            beforeEach(() => {
+                originalFinalizationRegistry = FinalizationRegistry as any
+                window.FinalizationRegistry = function (cleanupCb: (heldValue: any) => void) {
+                    const heldValueSet = new Map()
+
+                    cleanupMemory = ref => {
+                        const heldValue = heldValueSet.get(ref)
+
+                        cleanupCb(heldValue)
+                    }
+
+                    return ({
+                        register(target: object, heldValue: any, unregisterToken?: object) {
+                            heldValueSet.set(target, heldValue)
+                        },
+                        unregister() {}
+                    } as unknown) as FinalizationRegistry
+                } as any
+            })
+
+            afterEach(() => {
+                window.FinalizationRegistry = originalFinalizationRegistry as any
+            })
+
+            it('should be auto cleaned when ref in Map is freeing in program', function () {
+                const iwm = new IterableWeakMap()
+                const ref = {}
+                const value = {}
+                const ref2 = {}
+                const value2 = {}
+
+                iwm.set(ref, value)
+                iwm.set(ref2, value2)
+
+                expect(iwm.size).toBe(2)
+                expect(Array.from(iwm.entries())).toEqual([
+                    [ref, value],
+                    [ref2, value2]
+                ])
+
+                // emulate freeing "ref" in memory
+                cleanupMemory(ref)
+
+                expect(iwm.size).toBe(1)
+                expect(Array.from(iwm.entries())).toEqual([[ref2, value2]])
+            })
+        })
+    })
+
+    describe("environment doesn't support WeakRef and FinalizationRegistry", () => {
+        let originalWeakRef: WeakRefConstructor | undefined
+        let originalFinalizationRegistry: FinalizationRegistryConstructor | undefined
+
+        beforeAll(() => {
+            originalWeakRef = global.WeakRef
+            // @ts-ignore
+            global.WeakRef = undefined
+            originalFinalizationRegistry = global.FinalizationRegistry
+            // @ts-ignore
+            global.FinalizationRegistry = undefined
```

Comment:
Let's use `delete` instead of setting `undefined`.
Older or unsupported browser will not have it as a reference at all, so even expressions like `WeakRef ? A : B` won't work.
We should do the feature check with `typeof WeakRef !== 'undefined'`

Reply:
It's a very good point, changed to `typeof`